routes/api/category-routes.js

```diff
@@ -0,0 +1,77 @@
+const router = require('express').Router();
+const { Category, Product } = require('../../models');
+
+// The `/api/categories` endpoint
+
+router.get('/', async (req, res) => {
+  try {
+    const categoryData = await Category.findAll();
+    res.status(200).json(categoryData);
+  } catch (err) {
+    res.status(500).json(err);
+  }
+  // be sure to include its associated Products
+});
+
+router.get('/:id', async (req, res) => {
+  const id = req.params.id
+  // find one category by its `id` value
+  // be sure to include its associated Products
+  try {
+    const categoryData = await Category.findByPk(id);
+    if (!categoryData) {
+      res.status(404).json({ message: 'No Category found with this id!' });
+      return;
+    }
+
+    res.status(200).json(categoryData);
+  } catch (err) {
+    res.status(500).json(err);
+  }
+});
+
+router.post('/', async (req, res) => {
+  try {
+    const categoryData = await Category.create(req.body);
+    res.status(200).json(categoryData);
+  } catch (err) {
+    res.status(400).json(err);
+  }
+});
+
+router.put('/:id', async (req, res) => {
+  const id = req.params.id
+  try {
+    const categoryData = await Category.update(req.body, {
+      where: {
+        id,
+      },
+    })
+    res.status(200).json(categoryData);
+  } catch (err) {
+    res.status(400).json(err);
+  }
+  // update a category by its `id` value
+});
+
+router.delete('/:id', async (req, res) => {
+  const id = req.params.id
+  try {
+    const categoryData = await Category.destroy({
+      where: {
+        id
+      }
+    });
+
+    if (!categoryData) {
+      res.status(404).json({ message: 'No Category found with this id!' });
+      return;
+    }
+
+    res.status(200).json(categoryData);
```

Comment:
consider returning a confirmation message here as 'category' comes back as '1' (the number of rows deleted). It might be nicer for the user to receive something telling them the operation was successful instead :-)
Same for update routes.